content/post/2021-06-28-draw-me-a-project/index.md

```diff
@@ -0,0 +1,267 @@
+
+In the `blop/` folder there's a general orderly configuration that I haven't needed to touch, `oderly_config.yml".` There's also a `src/` folder corresponding to the source of my example report.
+
+    blop
+    ├── orderly_config.yml
+    └── src
+        └── example
+            ├── orderly.yml
+            └── script.R
+
+That example report consists of two files so it's a very small one. You could put anything in there as orderly is not opinionated about that! It *is* opinionated about being told about packages, resources, scripts, artefacts in the `blop/src/example/orderly.yml` configuration.
+
+In the example the configuration lists one script and two artefacts:
+
+``` yaml
+script: script.R
+
+artefacts:
+  - staticgraph:
+      description: A graph of things
+      filenames: mygraph.png
+  - data:
+      description: Data that went into the plot
+      filenames: mydata.csv
+```
+
+The script creates those two artefacts:
+
+``` r
+dat <- data.frame(x = 1:10, y = runif(10))
+write.csv(dat, "mydata.csv", row.names = FALSE)
+
+png("mygraph.png")
+plot(dat)
+dev.off()
+```
+
+Now how do you run the project?
+
+1.  You can use the [development mode](https://www.vaccineimpact.org/orderly/articles/orderly.html#developing-a-report-1) when developing a report, to have things at your current working directory.
+2.  You can build a draft version of the report with `id <- orderly::orderly_run("example", root = "blop")` after which your whole analysis, input and output, appears in `blop/draft/example/some-id-that-contains-the-date-and-a-hash`.
+3.  If you like it you can commit it with `orderly::orderly_commit(id, root = "blop")` which moves the whole analysis folder from `blop/draft/example/some-id-that-contains-the-date-and-a-hash` to `blop/archive/example/some-id-that-contains-the-date-and-a-hash`.
+
+Note that as the draft and archive folders can be gigantic you are expected to back up with some system. You are not expected to use git as it does not behave well with very large files.
+
+To get started with orderly you can read the [orderly website](https://www.vaccineimpact.org/orderly/) that I found very clear, and you should start with a small project... which again is my current non-expertise level.
+
+To follow orderly news you can
+
+-   [Release-watch](https://github.blog/changelog/2018-11-27-watch-releases/) the [orderly GitHub repo](https://github.com/vimc/orderly);
+-   read [the blog of the team behind orderly](https://reside-ic.github.io/);
+-   follow [Rich FitzJohn](https://twitter.com/rgfitzjohn) on Twitter.
+
+### Other tools for building analyses
+
+The targets and orderly packages are not the only ones helping run analyses. You might even want to build your own!
```

Comment:
No more to link too ? 

I know about at least this one which is cool : https://jdblischak.github.io/workflowr/

And this one too I saw mentioned once: https://github.com/nt-williams/cabinets

It is possible that they are all there: https://cran.r-project.org/web/views/ReproducibleResearch.html if you want to link to that.

Reply:
excellent point!